### ISSUE TITLE
Failed to read the 'localStorage' property from 'Window': Access is denied for this document

### ISSUE THREAD

**Issue by pokaalinkanssayohon** [MarketSquare/robotframework-browser#1612]
**Describe the bug**
When calling keyword 'Local Storage Set Item' => Failed to read the 'localStorage' property from 'Window': Access is denied for this document. Is this behind some Playwright / Browser setting?

**To Reproduce**
1. New Browser
2. New Context
3. Local Storage Set Item    <key>    <value>
4. See error: Failed to read the 'localStorage' property from 'Window': Access is denied for this document

**Expected behavior**
I try to set cookie value to local storage (needed to avoid cookie popup)

**Screenshots**

**Desktop (please complete the following information):**
 - OS: MacOS and Linux
 - Browser: chrome
 - PW browser of robotframework-browser          10.0.3

**Smartphone (please complete the following information):**

**Additional context**

**Comment by aaltat** [comment]
Would you be able to enable the Playwright debug logs from library import https://marketsquare.github.io/robotframework-browser/Browser.html#Importing and show us the log. Please note that the log contains everything as plain text.

**Comment by pokaalinkanssayohon** [comment]
Sure, I also updated rfbrowser to latest stable v 11.0.0 and re-executed the "case".
[playwright-log.txt](https://github.com/MarketSquare/robotframework-browser/files/7692246/playwright-log.txt)
.

**Comment by aaltat** [comment]
The `about:blank` page does have or does not grant access the local storage.

**Comment by pokaalinkanssayohon** [comment]
> The `about:blank` page does have or does not grant access the local storage.

Do you mean, that I should call New Page with url 'about:blank' instead of leaving url argument empty or how?

**Comment by aaltat** [comment]
I mean that you should navigate to real and live webpage page.

**Comment by pokaalinkanssayohon** [comment]
Ok, thanks, I will try this by using real working url in New Page -keyword. Interesting, that this local Storage works like this in Playwright/Browser setup 🤔. Thanks @aaltat for your effort 👍

**Comment by aaltat** [comment]
You may have been affected on automatic closing, have you seen this: https://marketsquare.github.io/robotframework-browser/Browser.html#Automatic%20page%20and%20context%20closing ?

**Comment by pokaalinkanssayohon** [comment]
Yes I have seen this. This shouldn't be the case, I have default value (=TEST) and this happens during one test case.

**Comment by xylix** [comment]
We could maybe look into having some way to pre-set localstorage / cookies when using `New Page` or something, so we can avoid this problem 🤔 . I think currently one work aroudn for this could be


 1.   New Browser
 2.  New Context
 3.  New Page  'url'
 4.  Local Storage Set Item 'key' 'value'
 5.  Go To  'url'  # To reset the page

@pokaalinkanssayohon if you have time to try this please let us know if it works

**Comment by pokaalinkanssayohon** [comment]
Thanks @xylix for workaround, this works!
Actually steps 3 and 5 has this same url and it works smoothly like that.

**Comment by pokaalinkanssayohon** [comment]
I think 'Go To' keyword might reset the cookie setting as well (?). Could that be the reason?

**Comment by xylix** [comment]
I think `Go To` should not reset it. Seems like a bug.

I'm unsure though what sort of fix would be good for this use case here. It would be difficult to have `Local Storage set` keywords work on an empty page with our current implementation.

@pokaalinkanssayohon Sorry for the delay, had some christmas holidays 😅 . I did come up with a new idea for a work around though:

In suite setup or some other place which is run before tests:

1. New Browser
2. New Context
3. New Page 'url'
4. Local Storage Set Item 'key' 'value'
5. ${state_file} =    Save Storage State

Then in the real tests:
1. New Context  storageState=${state_file}
2. New Page  'url'

**Comment by pokaalinkanssayohon** [comment]
Hi @xylix ! I tried this workaround but it didn't help either, unfortunately. Cookie values seems to be stored locally but still cookie accept popup appears.

**Comment by xylix** [comment]
Dang. I will try to think either new feature to solve this or a workaround. Also asking the team for help.
